framework/io/obj_io.cc

```diff
@@ -0,0 +1,392 @@
+    {
+      Edge edge;
+      for (int k = 1; k <= 2; ++k)
+      {
+        // Extract sub word
+        beg_of_word = file_line.find_first_not_of(delimiter, end_of_word);
+        end_of_word = file_line.find(delimiter, beg_of_word);
+        sub_word = file_line.substr(beg_of_word, end_of_word - beg_of_word);
+
+        // Convert word to number
+        try
+        {
+          int vertex_id = std::stoi(sub_word);
+          if (k == 1)
+            edge.first = vertex_id - 1;
+          if (k == 2)
+            edge.second = vertex_id - 1;
+        }
+
+        // Catch conversion error
+        catch (const std::invalid_argument& ia)
+        {
+          log.Log0Warning() << "Failed to text to integer in line " << file_line << std::endl;
+        }
+      } // for k
+
+      if (block_data.empty())
+        throw std::logic_error(fname + ": Could not add edge to block-data. "
+                                       "This normally indicates that the file does not have the "
+                                       "\"o Object Name\" entry.");
+
+      block_data.back().edges.push_back(edge);
+    } // if (first_word == "l")
+  }
+  file.close();
+  log.Log0Verbose0() << "Max material id: " << material_id;
+
+  // Error checks
+  for (const auto& block : block_data)
+    for (const auto& cell : block.cells)
+      for (const auto vid : cell->vertex_ids)
+      {
+        OpenSnLogicalErrorIf(
+          vid >= file_vertices.size(),
+          "Cell vertex id " + std::to_string(vid) +
+            " not in list of vertices read (size=" + std::to_string(file_vertices.size()) + ").");
+      }
+
+  // Filter blocks
+  std::vector<size_t> bndry_block_ids;
+  size_t num_cell_blocks = 0;
+  size_t main_block_id = 0;
+  for (size_t block_id = 0; block_id < block_data.size(); ++block_id)
+  {
+    if (not block_data[block_id].edges.empty())
+      bndry_block_ids.push_back(block_id);
+    if (not block_data[block_id].cells.empty())
+    {
+      ++num_cell_blocks;
+      main_block_id = block_id;
+    }
+  } // for block_id
+
+  if (num_cell_blocks != 1)
+    throw std::logic_error(fname +
+                           ": More than one cell-block has been read "
+                           "from the file. Only a single face-containing object is supported. "
+                           "If you exported this mesh from blender, be sure to export "
+                           "\"selection only\"");
+
+  // Process blocks
+  std::vector<Vertex> cell_vertices;
+  {
+    // Initial map is straight
+    std::vector<size_t> vertex_map;
+    vertex_map.reserve(file_vertices.size());
+    for (size_t m = 0; m < file_vertices.size(); ++m)
+      vertex_map.push_back(m);
+
+    // Build set of cell vertices
+    std::set<size_t> cell_vertex_id_set;
+    for (const auto& cell_ptr : block_data.at(main_block_id).cells)
+      for (size_t vid : cell_ptr->vertex_ids)
+        cell_vertex_id_set.insert(vid);
+
+    // Make cell_vertices and edit map
+    {
```

Comment:
Why do we restrict scope here?

Reply:
I think somebody wanted to use range-based for loop, but they needed an iteration index, but did not want to propagate the name further in the code... This is just 🤦‍♂️

Reply:
nvm, `cell_vertex_id_set` is a `std::set`, so only to restrict the scope of the `new_id`.